traffic_ops/testing/api/v3/servers_test.go

```diff
@@ -166,24 +166,54 @@ func GetTestServersQueryParameters(t *testing.T) {
 
 	params := url.Values{}
 	params.Add("dsId", strconv.Itoa(*ds.ID))
-	_, _, err = TOSession.GetServers(&params)
+	_, _, err = TOSession.GetServersWithHdr(&params, nil)
 	if err != nil {
 		t.Fatalf("Failed to get server by Delivery Service ID: %v", err)
 	}
 
 	currentTime := time.Now().UTC().Add(5 * time.Second)
-	time := currentTime.Format(time.RFC1123)
+	timestamp := currentTime.Format(time.RFC1123)
 	var header http.Header
 	header = make(map[string][]string)
-	header.Set(rfc.IfModifiedSince, time)
+	header.Set(rfc.IfModifiedSince, timestamp)
 	_, reqInf, _ := TOSession.GetServersWithHdr(&params, header)
 	if reqInf.StatusCode != http.StatusNotModified {
 		t.Errorf("Expected a status code of 304, got %v", reqInf.StatusCode)
 	}
 
+	foundTopDs := false
+	const topDsXmlId = "ds-top"
+	for _, ds = range dses {
+		if ds.XMLID != nil && *ds.XMLID != topDsXmlId {
+			continue
+		}
+		foundTopDs = true
+		break
```

Comment:
idk if this logic makes sense. So if you get a Delivery Service with a nil XMLID it's the `ds-top` DS?

Reply:
Oops, I just mindlessly added a nil check. Logic fixed in 488f88fb8